src/meta_schedule/space_generator/post_order_apply.cc

```diff
@@ -106,20 +95,63 @@ class PostOrderApplyNode : public SpaceGeneratorNode {
 
   Array<tir::Schedule> GenerateDesignSpace(const IRModule& mod_) final {
     using ScheduleAndUnvisitedBlocks = std::pair<tir::Schedule, Array<tir::BlockRV>>;
-    tir::Schedule sch = tir::Schedule::Traced(        //
-        /*mod=*/mod_,                                 //
-        /*rand_state=*/ForkSeed(&this->rand_state_),  //
-        /*debug_mode=*/0,                             //
+    tir::Schedule sch = tir::Schedule::Traced(                          //
+        /*mod=*/mod_,                                                   //
+        /*rand_state=*/ForkSeed(&this->rand_state_),                    //
+        /*debug_mode=*/tir::kVerifySRefTree | tir::kVerifyCachedFlags,  //
         /*error_render_level=*/tir::ScheduleErrorRenderLevel::kDetail);
 
     std::vector<ScheduleAndUnvisitedBlocks> stack;
-    Array<tir::Schedule> result{sch};
+    Array<tir::Schedule> result;
+    Array<tir::BlockRV> all_blocks = BlockCollector::Collect(sch), func_blocks, non_func_blocks;
+    for (const tir::BlockRV& block_rv : all_blocks) {
+      if (tir::GetAnn<String>(sch->GetSRef(block_rv), "schedule_rule")) {
+        func_blocks.push_back(block_rv);
+      } else {
```

Comment:
We can preemptively filter out the "None" here. What do you think?

```suggestion
      if (Optional<String> rule = tir::GetAnn<String>(sch->GetSRef(block_rv), "schedule_rule")) {
        if (rule.value() != "None") {
          func_blocks.push_back(block_rv);
        }
      } else {
```